src/query_plan/operation.rs

```diff
@@ -759,6 +744,13 @@ pub(crate) mod normalized_field_selection {
         pub(crate) selection_set: Option<NormalizedSelectionSet>,
     }
 
+    impl NormalizedFieldSelection {
```

Comment:
can we move all those selections `has_defer`s to top level impls? per previous review comments the inner mods created to keep the key/value in sync but all the other stuff should be moved outside